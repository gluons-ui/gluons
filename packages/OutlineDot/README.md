## Mandala UI OutlineDot Component
Mandala is a React component library that utilizes a declarative CSS library for styling.

### Installation
Check the [MonoRepo README](https://github.com/mandala-ui/mandala) for installation.

### Description
A header for the top of lists that matches the style and props

### Usage
if using the whole library:
```js
import { OutlineDot } from 'mandala';
```
if only using this component:
```js
import OutlineDot from '@mandala-ui/outline-dot';
```
### Props
* `borderColor` - the color of the border around the dot
* `borderWidth` - the width of the border around the dot
* `color` - the color of the dot
* `innerBorderWidth` - the width of the inner section between the dot and border
* `innerColor` - the color of the inner section of the outlined dot

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| borderColor | string| null         | -          |
| borderWidth | number| false        | -          |
| color       | string | 'white'     | -          |
| innerBorderWidth | number  | false | -          |
| innerColor  | string | 0           | -          |
| large       | boolean| false       | -          |

#### TODO:
- [ ] rename to Dot, use outline as a prop
- [ ] infer `lined` from lineColor prop existence

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
