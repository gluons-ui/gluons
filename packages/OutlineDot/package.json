{
  "author": "Lucas K Allmon",
  "description": "Mandala UI Kit OutlineDot Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user"
  ],
  "license": "MIT",
  "main": "lib/OutlineDot.js",
  "module": "es/OutlineDot.js",
  "name": "@mandala-ui/outline-dot",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.1"
}
