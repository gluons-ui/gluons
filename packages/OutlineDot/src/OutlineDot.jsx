import React from 'react';
import PropTypes from 'prop-types';

const OutlineDot = ({
  borderColor,
  borderWidth,
  color,
  innerBorderWidth,
  innerColor,
  large }) => {
  const isLarge = `${large ? 'h2 w2' : 'h1 w1'}`;
  const outerBorder = `${borderWidth === 0 ? '' : `bw${borderWidth}`}`;
  const innerBorder = `${innerBorderWidth === 0 ? '' : `bw${innerBorderWidth}`}`;
  return (
    <div className={`br-100 flex items-center justify-center ${isLarge}`}>
      <span className={`b--${borderColor} ba ${outerBorder} br-100 flex items-center justify-center ${isLarge}`}>
        <div className={`ba bg-${color} br-100 b--${innerColor} ${innerBorder} ${isLarge}`} />
      </span>
    </div>
  );
};

OutlineDot.propTypes = {
  borderColor: PropTypes.string,
  borderWidth: PropTypes.number,
  innerColor: PropTypes.string,
  color: PropTypes.string,
  innerBorderWidth: PropTypes.number,
  large: PropTypes.bool,
};

OutlineDot.defaultProps = {
  borderColor: 'black',
  borderWidth: 0,
  innerBorderWidth: 1,
  innerColor: 'white',
  color: 'black',
  large: false,
};

export default OutlineDot;
