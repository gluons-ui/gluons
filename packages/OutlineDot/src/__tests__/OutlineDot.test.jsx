import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import OutlineDot from '../OutlineDot';

configure({ adapter: new Adapter() });

describe('<OutlineDot />', () => {
  describe('Rendering', () => {
    it('renders Default', () => {
      expect(shallow(
        <OutlineDot />,
      )).toMatchSnapshot();
    });
    it('renders a different color outline', () => {
      expect(shallow(
        <OutlineDot
          borderColor="blue"
        />,
      )).toMatchSnapshot();
    });
    it('renders a thicker border', () => {
      expect(shallow(
        <OutlineDot
          borderWidth={2}
        />,
      )).toMatchSnapshot();
    });
    it('renders a thicker inner area', () => {
      expect(shallow(
        <OutlineDot
          innerBorderWidth={2}
        />,
      )).toMatchSnapshot();
    });
    it('renders a different color inner area', () => {
      expect(shallow(
        <OutlineDot
          innerColor="red"
        />,
      )).toMatchSnapshot();
    });
    it('renders a different color dot', () => {
      expect(shallow(
        <OutlineDot
          color="blue"
        />,
      )).toMatchSnapshot();
    });
    it('renders no borders', () => {
      expect(shallow(
        <OutlineDot
          innerBorderWidth={0}
          borderWidth={0}
        />,
      )).toMatchSnapshot();
    });
    it('renders a large version', () => {
      expect(shallow(
        <OutlineDot
          large
        />,
      )).toMatchSnapshot();
    });
    it('renders no borders and large', () => {
      expect(shallow(
        <OutlineDot
          borderColor="green"
          borderWidth={0}
          color="green"
          innerColor="white"
          innerBorderWidth={0}
          large
        />,
      )).toMatchSnapshot();
    });
  });
});
