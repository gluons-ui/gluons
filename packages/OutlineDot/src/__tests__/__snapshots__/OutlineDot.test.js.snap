// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutlineDot /> Rendering renders Default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="black"
    innerBorderWidth={1}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white bw1 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white bw1 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a different color dot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="blue"
    innerBorderWidth={1}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-blue br-100 b--white bw1 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-blue br-100 b--white bw1 h1 w1"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-blue br-100 b--white bw1 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-blue br-100 b--white bw1 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-blue br-100 b--white bw1 h1 w1"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-blue br-100 b--white bw1 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a different color inner area 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="black"
    innerBorderWidth={1}
    innerColor="red"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--red bw1 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--red bw1 h1 w1"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--red bw1 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--red bw1 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--red bw1 h1 w1"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--red bw1 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a different color outline 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="blue"
    borderWidth={0}
    color="black"
    innerBorderWidth={1}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--blue ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />,
        "className": "b--blue ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white bw1 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--blue ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />,
          "className": "b--blue ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white bw1 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a large version 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="black"
    innerBorderWidth={1}
    innerColor="white"
    large={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h2 w2"
      >
        <div
          className="ba bg-black br-100 b--white bw1 h2 w2"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h2 w2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white bw1 h2 w2"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h2 w2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white bw1 h2 w2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h2 w2"
        >
          <div
            className="ba bg-black br-100 b--white bw1 h2 w2"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h2 w2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white bw1 h2 w2"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h2 w2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white bw1 h2 w2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a thicker border 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={2}
    color="black"
    innerBorderWidth={1}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba bw2 br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white bw1 h1 w1"
        />,
        "className": "b--black ba bw2 br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white bw1 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba bw2 br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white bw1 h1 w1"
          />,
          "className": "b--black ba bw2 br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white bw1 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders a thicker inner area 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="black"
    innerBorderWidth={2}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--white bw2 h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white bw2 h1 w1"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white bw2 h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--white bw2 h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white bw2 h1 w1"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white bw2 h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders no borders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="black"
    borderWidth={0}
    color="black"
    innerBorderWidth={0}
    innerColor="white"
    large={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--black ba  br-100 flex items-center justify-center h1 w1"
      >
        <div
          className="ba bg-black br-100 b--white  h1 w1"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h1 w1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-black br-100 b--white  h1 w1"
        />,
        "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-black br-100 b--white  h1 w1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--black ba  br-100 flex items-center justify-center h1 w1"
        >
          <div
            className="ba bg-black br-100 b--white  h1 w1"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h1 w1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-black br-100 b--white  h1 w1"
          />,
          "className": "b--black ba  br-100 flex items-center justify-center h1 w1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-black br-100 b--white  h1 w1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<OutlineDot /> Rendering renders no borders and large 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutlineDot
    borderColor="green"
    borderWidth={0}
    color="green"
    innerBorderWidth={0}
    innerColor="white"
    large={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="b--green ba  br-100 flex items-center justify-center h2 w2"
      >
        <div
          className="ba bg-green br-100 b--white  h2 w2"
        />
      </span>,
      "className": "br-100 flex items-center justify-center h2 w2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="ba bg-green br-100 b--white  h2 w2"
        />,
        "className": "b--green ba  br-100 flex items-center justify-center h2 w2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "ba bg-green br-100 b--white  h2 w2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="b--green ba  br-100 flex items-center justify-center h2 w2"
        >
          <div
            className="ba bg-green br-100 b--white  h2 w2"
          />
        </span>,
        "className": "br-100 flex items-center justify-center h2 w2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ba bg-green br-100 b--white  h2 w2"
          />,
          "className": "b--green ba  br-100 flex items-center justify-center h2 w2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ba bg-green br-100 b--white  h2 w2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
